presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -187,6 +187,9 @@ public long getLong()
             if (value instanceof Long || value instanceof Integer) {
                 return ((Number) value).longValue();
             }
+            else if (value instanceof Float) {
+                return floatToIntBits((Float) value) & 0xFF_FF_FF_FFL;
```

Comment:
the mask is not needed here, is it?

Reply:
Thanks:) It's not needed, will update.

Reply:
It's actually wrong - if value is negative, so definitely removing it.